sdk/cosmos/azure_data_cosmos/src/options/cosmos_client_options.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::{builders::ClientOptionsBuilder, ClientOptions};
+
+/// Options used when creating a [`CosmosClient`](crate::CosmosClient).
+#[derive(Clone, Debug, Default)]
+pub struct CosmosClientOptions {
+    pub(crate) client_options: ClientOptions,
+}
+
+impl CosmosClientOptions {
+    /// Creates a new [`CosmosClientOptionsBuilder`](CosmosClientOptionsBuilder) that can be used to construct a [`CosmosClientOptions`].
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// let options = azure_data_cosmos::ReadDatabaseOptions::builder().build();
+    /// ```
+    pub fn builder() -> CosmosClientOptionsBuilder {
+        CosmosClientOptionsBuilder::default()
+    }
+}
+
+/// Builder used to construct a [`CosmosClientOptions`].
+///
+/// Obtain a [`CosmosClientOptionsBuilder`] by calling [`CosmosClientOptions::builder()`]
+#[derive(Default)]
+pub struct CosmosClientOptionsBuilder(CosmosClientOptions);
+
+impl CosmosClientOptionsBuilder {
+    /// Builds a [`CosmosClientOptions`] from the builder.
+    ///
+    /// This does not consume the builder, and can be called multiple times.
+    pub fn build(&self) -> CosmosClientOptions {
+        self.0.clone()
```

Comment:
`.0`/`.1`/etc is the syntax for accessing items in a tuple. So `("a", "b", "c").1 == "b"`, for example.

The `CosmosClientOptionsBuilder` is a ["tuple-struct"](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-tuple-structs-without-named-fields-to-create-different-types), which is a common pattern to use when you have a struct with only one member (though you can have multiple tuple members, i.e. `pub struct Foo(u32, String, ...)` which would be addressed as `self.0`, `self.1`, etc.)

So, this is accessing the `CosmosClientOptions` that is the only field within the `CosmosClientOptionsBuilder`, and then cloning it to return to the user.

Reply:
Thanks for explaining!  I had used tuples in rust, but hadn't done "tuple-structs".

Reply:
Personally, I dislike tuple-structs when the contained number of values is > 1 because remembering which is 0, 1, etc., it's a headache. That said, I feel no reason to have a guideline about this for Azure SDKs. Yet. If people start abusing it (say, 3 or more and often) we might have to because it gets hard to review and maintain.